File: Python/2.两数相加.py
#
# @lc app=leetcode.cn id=2 lang=python3
#
# [2] 两数相加
#
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def addTwoNumbers(self, l1: ListNode, l2: ListNode) -> ListNode:
        l1_num = list()
        l2_num = list()
        while l1:
            l1_num.append(str(l1.val))
            l1 = l1.next
        while l2:
            l2_num.append(str(l2.val))
            l2 = l2.next
        l1_num = int(''.join(l1_num)[::-1])
        l2_num = int(''.join(l2_num)[::-1])
        res_num = str(l1_num + l2_num)[::-1]
        res = ListNode(int(res_num[0]))
        tmp = res
        for num in res_num[1:]:
            tmp.next = ListNode(int(num))
            tmp = tmp.next
        return res
        

File: 2.两数相加.java
/*
 * @lc app=leetcode.cn id=2 lang=java
 *
 * [2] 两数相加
 */

// @lc code=start
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
        ListNode res = new ListNode(0);
        ListNode tmp = res;
        int flag = 0;
        while(l1!=null || l2!=null){
            int x = (l1 != null) ? l1.val : 0;
            int y = (l2 != null) ? l2.val : 0;
            int sum = x + y + flag;
            flag = sum / 10;
            tmp.next = new ListNode(sum%10);
            if (l1!=null) l1 = l1.next;
            if (l2!=null) l2 = l2.next;
            tmp = tmp.next;
        }
        if(flag==1){
            tmp.next = new ListNode(flag);
        }
        return res.next;
    }
}
// @lc code=end

